pkg/cdi/spec.go

```diff
@@ -0,0 +1,170 @@
+package cdi
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+
+	oci "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/yaml"
+
+	specs "github.com/container-orchestrated-devices/container-device-interface/specs-go"
+)
+
+var (
+	// ValidSpecVersions contain the valid CDI Spec versions.
+	ValidSpecVersions = map[string]struct{}{
+		"":      {},
```

Comment:
Good point. Since this seems to be modeled closely after the OCI Spec, probably an empty version is not something that we're keen to allow. I'll remove that.

Reply:
One (unrelated) question I had was how we expose which version of the spec is present in the package. Do we expect users to set this manually, or do we have a `const` somewhere?